src/main/java/org/glassfish/grizzly/memcached/jmx/GrizzlyMemcachedCache.java

```diff
@@ -0,0 +1,245 @@
+    @ParameterNames({"hostname","port"})
+    public int getActiveCount(final String hostname, final int port) {
+        return cache.getActiveCount(getAddress(hostname, port));
+    }
+
+    @ManagedOperation(id = "grizzly-memcached-cache-idle-count", impact = Impact.INFO)
+    @Description("The number of connections currently idle in this pool or a negative value if unsupported.")
+    @ParameterNames({"hostname","port"})
+    public int getIdleCount(final String hostname, final int port) {
+        return cache.getIdleCount(getAddress(hostname, port));
+    }
+
+    @ManagedOperation(id = "grizzly-memcached-connection-stat", impact = Impact.INFO)
+    @Description("The stat of connections in this pool.")
+    @ParameterNames({"hostname","port"})
+    public CompositeConnectionStat getConnectionStat(final String hostname, final int port) {
+        final SocketAddress server = getAddress(hostname, port);
+        return new CompositeConnectionStat(cache.getConnectionSize(server), cache.getPeakCount(server),
+                                           cache.getActiveCount(server), cache.getIdleCount(server));
+    }
+
+    @ManagedAttribute(id = "grizzly-memcached-cache-jmx-enabled")
+    public boolean isJmxEnabled() {
+        return cache.isJmxEnabled();
+    }
+
+    private void rebuildSubTree() {
+        final Transport transport = cache.getTransport();
+        if (currentTransport != transport) {
+            if (currentTransport != null) {
+                mom.deregister(transportJmx);
+                currentTransport = null;
+                transportJmx = null;
+            }
+            if (transport != null) {
+                final Object jmx = transport.getMonitoringConfig().createManagementObject();
+                mom.register(this, jmx);
+                currentTransport = transport;
+                transportJmx = jmx;
+            }
+        }
+
+        final ObjectPool connectionPool = cache.getConnectionPool();
+        if (currentConnectionPool != connectionPool) {
+            if (currentConnectionPool != null) {
+                mom.deregister(connectionPoolJmx);
+                currentConnectionPool = null;
+                connectionPoolJmx = null;
+            }
+            if (connectionPool != null) {
+                final Object jmx = connectionPool.getMonitoringConfig().createManagementObject();
+                mom.register(this, jmx);
+                currentConnectionPool = connectionPool;
+                connectionPoolJmx = jmx;
+            }
+        }
+    }
+
+    private static SocketAddress getAddress(final String hostname, final int port) {
+        try {
+            return new InetSocketAddress(hostname, port);
+        } catch(Exception ignore) {
+            return null;
+        }
+    }
+
+    @ManagedData(name="Connection Stat")
+    private static class CompositeConnectionStat {
```

Comment:
This one is `private`, but `o.g.g.m.p.jmx.KeyedObject.CompositeObjectStat` and  `o.g.g.m.p.jmx.BaseObjectPool.CompositeObjectPoolStat` are `public`. If there is no specific reason - for consistency, could it be synchronized?

Reply:
Since it has no effect whether private class or public class, I synchronized it to private one.